server/src/lib.rs

```diff
@@ -58,12 +67,43 @@ use rand::{distributions::Alphanumeric, thread_rng, Rng};
 use regex::{Regex, RegexBuilder};
 use serde::Deserialize;
 
+pub type DbPool = diesel::r2d2::Pool<diesel::r2d2::ConnectionManager<diesel::PgConnection>>;
 pub type ConnectionId = usize;
 pub type PostId = i32;
 pub type CommunityId = i32;
 pub type UserId = i32;
 pub type IPAddr = String;
 
+#[derive(Clone, Debug, Fail)]
+#[fail(display = "Error sending request, {}", _0)]
+pub struct SendError(pub String);
+
```

Comment:
We have an issue to replace failure with anyhow and thiserror #689 . Different PR obvi, just a note.